remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/mappers/JsonRulesMapper.kt

```diff
@@ -0,0 +1,224 @@
+            max = jsonMatchingAttribute.max.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            fallback = jsonMatchingAttribute.fallback
+        )
+    }
+
+    private val favoritesMapper: (JsonMatchingAttribute) -> MatchingAttribute = { jsonMatchingAttribute ->
+        MatchingAttribute.Favorites(
+            value = jsonMatchingAttribute.value.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            min = jsonMatchingAttribute.min.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            max = jsonMatchingAttribute.max.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            fallback = jsonMatchingAttribute.fallback
+        )
+    }
+
+    private val appThemeMapper: (JsonMatchingAttribute) -> MatchingAttribute = { jsonMatchingAttribute ->
+        MatchingAttribute.AppTheme(
+            value = jsonMatchingAttribute.value.toStringOrDefault(MATCHING_ATTR_STRING_DEFAULT_VALUE),
+            fallback = jsonMatchingAttribute.fallback
+        )
+    }
+
+    private val daysSinceInstalledMapper: (JsonMatchingAttribute) -> MatchingAttribute = { jsonMatchingAttribute ->
+        MatchingAttribute.DaysSinceInstalled(
+            value = jsonMatchingAttribute.value.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            min = jsonMatchingAttribute.min.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            max = jsonMatchingAttribute.max.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            fallback = jsonMatchingAttribute.fallback
+        )
+    }
+
+    private val daysUsedSinceMapper: (JsonMatchingAttribute) -> MatchingAttribute = { jsonMatchingAttribute ->
+        MatchingAttribute.DaysUsedSince(
+            since = dateFormatter.parse(jsonMatchingAttribute.since as String)!!,
+            value = jsonMatchingAttribute.value.toIntOrDefault(MATCHING_ATTR_INT_DEFAULT_VALUE),
+            fallback = jsonMatchingAttribute.fallback
+        )
+    }
+
+    private val attributesMappers = mapOf(
+        Pair("locale", localeMapper),
+        Pair("osApi", osApiMapper),
+        Pair("webview", webViewMapper),
+        Pair("flavor", flavorMapper),
+        Pair("appId", appIdMapper),
+        Pair("appVersion", appVersionMapper),
+        Pair("atb", atbMapper),
+        Pair("appAtb", appAtbMapper),
+        Pair("searchAtb", searchAtbMapper),
+        Pair("expVariant", expVariantMapper),
+        Pair("installedGPlay", installedGPlayMapper),
+        Pair("defaultBrowser", defaultBrowserMapper),
+        Pair("emailEnabled", emailEnabledMapper),
+        Pair("widgetAdded", widgetAddedMapper),
+        Pair("searchCount", searchCountMapper),
+        Pair("bookmarks", bookmarksMapper),
+        Pair("favorites", favoritesMapper),
+        Pair("appTheme", appThemeMapper),
+        Pair("daysSinceInstalled", daysSinceInstalledMapper),
+        Pair("daysUsedSince", daysUsedSinceMapper)
+    )
+
+    fun map(jsonRules: List<JsonMatchingRule>): Map<Int, List<MatchingAttribute>> = jsonRules
```

Comment:
I would make this an extension fun of `List<JsonMatchingRule>` type, removing the wrapper `JsonRulesMapper` class